pkgs/applications/window-managers/i3/wmfocus.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, fetchFromGitHub, rustPlatform,
+  xorg, python3, pkgconfig, glib, cmake, cairo, libxkbcommon }:
+let
+  pname = "wmfocus";
+  version = "1.0.2";
+in
+rustPlatform.buildRustPackage {
+  inherit pname version;
+  name = "${pname}-${version}";
+
+  buildInputs = [python3 pkgconfig glib xorg.libX11 cmake cairo xorg.libX11];
+  nativeBuildInputs = [libxkbcommon xorg.libxcb xorg.xcbutilkeysyms];
+
```

Comment:
These `buildInputs` and `nativeBuildInputs` don't look right. `nativeBuildInputs` should only contain packages that need to run at buildtime (here: `pkgconfig` and `python3`). The rest should be `buildInputs`. Some of them, like `cmake` aren't needed at all. I found this to be sufficient to build the package (and it seems to run fine):

```
nativeBuildInputs = [ python3 pkgconfig ];
buildInputs = [ cairo libxkbcommon xorg.xcbutilkeysyms ];
```

These `buildInputs` also match the upstream list of dependencies (minus rust and cargo, which are implied by buildRustPackage).

Reply:
@xeji 2255419